controllers/workspace/status.go

```diff
@@ -165,6 +167,9 @@ func getInfoMessage(workspace *devworkspace.DevWorkspace, conditions map[devwork
 	if _, ok := conditions[devworkspace.WorkspaceServiceAccountReady]; ok {
 		return "Waiting for deployment to be ready"
 	}
+	if _, ok := conditions[PullSecretsReadyCondition]; ok {
+		return "Waiting for workspace pull secrets"
+	}
 	if _, ok := conditions[devworkspace.WorkspaceRoutingReady]; ok {
 		return "Waiting for workspace serviceaccount"
 	}
```

Comment:
Are we sure this is the right logic? I would assume that if the `PullSecretsReady` condition is true, then the pull secrets have been prepared and we're not actually waiting on them. 

The general approach to this progress message I've taken is once a condition is set, we're waiting on the next condition, i.e.
```suggestion
	if _, ok := conditions[devworkspace.WorkspaceServiceAccountReady]; ok {
		return "Waiting for deployment to be ready"
	}
	if _, ok := conditions[PullSecretsReadyCondition]; ok {
		return "Waiting for workspace service account"
	}
	if _, ok := conditions[devworkspace.WorkspaceRoutingReady]; ok {
		return "Waiting for workspace pull secrets"
	}
```
Each condition gets set in sequence, so the idea is once a condition is set, the workspace is waiting on the _next_ condition in the list -- If `conditions[devworkspace.WorkspaceServiceAccountReady]` is set, then we know `conditions[PullSecretsReadyCondition]` is set, etc.

Reply:
Good catch.
it's downside of multitasking :-D

I'll fix it

Reply:
I refactored the way how we expose info message https://github.com/devfile/devworkspace-operator/pull/284/commits/d208482adb745eb279aee3fb99747c70519d3672

And it seems to work fine but I will need to take a look why 
waiting for editor did not appeared, maybe it just passed to fast

```
> kc get dw theia -w
NAME    WORKSPACE ID                PHASE   INFO
theia   workspace60395b9042b94129           
theia   workspace60395b9042b94129           
theia   workspace60395b9042b94129           
theia   workspace60395b9042b94129   Starting   Processing DevWorkspace components
theia   workspace60395b9042b94129   Starting   Processing DevWorkspace components
theia   workspace60395b9042b94129   Starting   Waiting for workspace routing objects
theia   workspace60395b9042b94129   Starting   Waiting for workspace routing objects
theia   workspace60395b9042b94129   Starting   Waiting for workspace routing objects
theia   workspace60395b9042b94129   Starting   Waiting for workspace serviceaccount
theia   workspace60395b9042b94129   Starting   Waiting for workspace serviceaccount
theia   workspace60395b9042b94129   Starting   Waiting for workspace serviceaccount
theia   workspace60395b9042b94129   Starting   Waiting for deployment to be ready
theia   workspace60395b9042b94129   Starting   Waiting for deployment to be ready
theia   workspace60395b9042b94129   Starting   Waiting for deployment to be ready
theia   workspace60395b9042b94129   Running    https://workspace60395b9042b94129.apps.cluster-4237.4237.sandbox422.opentlc.com/theia/
theia   workspace60395b9042b94129   Running    https://workspace60395b9042b94129.apps.cluster-4237.4237.sandbox422.opentlc.com/theia/
theia   workspace60395b9042b94129   Running    https://workspace60395b9042b94129.apps.cluster-4237.4237.sandbox422.opentlc.com/theia/
theia   workspace60395b9042b94129   Running    https://workspace60395b9042b94129.apps.cluster-4237.4237.sandbox422.opentlc.com/theia/
```

Reply:
Sometimes it starts quickly :)